assets/js/modules/analytics-4/components/dashboard/ActivationBanner/SetupBanner.js

```diff
@@ -19,28 +19,41 @@
 /**
  * WordPress dependencies
  */
+import { Fragment } from '@wordpress/element';
 import { __, sprintf } from '@wordpress/i18n';
 
 /**
  * Internal dependencies
  */
 import Data from 'googlesitekit-data';
 import BannerNotification from '../../../../../components/notifications/BannerNotification';
-import { Grid, Row, Cell } from '../../../../../material-components';
+import {
+	PropertySelect,
+	UseSnippetSwitch,
+} from '../../../../analytics-4/components/common';
+import useExistingTagEffect from '../../../../analytics-4/hooks/useExistingTagEffect';
+import { MODULES_ANALYTICS } from '../../../../analytics/datastore/constants';
 import { MODULES_ANALYTICS_4 } from '../../../datastore/constants';
 const { useSelect } = Data;
 
 export default function SetupBanner( { onCTAClick } ) {
+	// We need this useSelect hook to trigger starting getAccounts resolution which is needed to properly
+	// display the Property Select.
+	useSelect( ( select ) => select( MODULES_ANALYTICS ).getAccounts() );
```

Comment:
Hmm, this is… kinda odd. I guess it makes sense because our selectors have side-effects and it's what we're relying on here… though it's a bit odd the data or `hasFinishedResolution` check isn't being used. If this is needed to render something in this component's output, why aren't we checking to make sure it's defined?

Or if another component needs this data, shouldn't _it_ be the one making sure this data is loaded?

Reply:
Excellent spot, @tofumatt! These were actually my questions as well when I couldn't make the `<PropertySelect />` show up without calling this selector.

I did quite some backtracking and [it looked like](https://github.com/google/site-kit-wp/commit/69408425b855f407d88ef1b23c85056ab2af8da3#diff-d7f93ad4ddb90e5e27d32343c264d4d6a6f4e50d7fea61c4bef0a9df85d81377R58-R71) this check for accounts in `PropertySelect` existed from the initiation of the component, but couldn't figure out a reason why it was actually needed. I even tried removing the condition and couldn't notice any errors/side effects.

But I didn't want to risk it and noticed a few other places where a similar implementation was done ([example](https://github.com/google/site-kit-wp/blob/a1c48f185466353a5296c8ef4a818125b0321785/assets/js/modules/search-console/components/settings/SettingsEdit.js#L31-L35)), which is why I just went with calling the selector like this.

Do you possibly have an idea why do we actually need this `getAccounts` selector call in `PropertySelect`?

Thank you!

Reply:
The reason for "needing" it seems to be this `isLoading` check:

https://github.com/google/site-kit-wp/blob/286fc782e27170b3af93f18b06ee8611a47c6094/assets/js/modules/analytics-4/components/common/PropertySelect.js#L69-L71

But it checks for the resolution without actually calling it. We _could_ do what you added and saw elsewhere in the repo, but that introduces selector dependencies for rendering a component which is actually really weird to me.

Let's instead just add the selector right into `PropertySelect`. It's harmless to do from my point-of-view and if the component considers it a required field, we should use it 🙂